README.md

```diff
@@ -43,11 +70,34 @@ As you go further with quantum development, we provide several different categor
 
 We also encourage taking a look at the [unit tests](./samples/tests) used to check the correctness of the Quantum Development Kit samples.
 
-## Docker image
+## Setting up your development environment ##
+
+This repo contains several configuration files that will make it easy to get started with coding. Below we lay out some instructions for getting started with [VSCode](#visual-studio-code) or [Visual Studio Codespaces](#visual-studio-codespaces). The latter will use the Docker image that has all dependencies pre-installed for you. If you prefer working with Jupyter notebooks, we also provide instructions for how to set that up [below](#running-a-jupyter-notebook-with-docker).
+
+### Visual Studio Codespaces ###
+
+For minimal set-up time, we recommend using [Visual Studio Codespaces](https://online.visualstudio.com/login) in the browser. To get started, perform the following steps:
+
+1. Create an Azure account [here](https://azure.microsoft.com/en-us/free/), if you don't already have one. You'll have to set up payment information - note that you get $200 of free credit and you won't be charged unless you upgrade.
+2. Create a Codespace environment. Please follow the quickstart guide [here](https://docs.microsoft.com/en-us/visualstudio/online/quickstarts/browser).
+When creating the Codespace, make sure to enter `microsoft/Quantum` in the "Git Repository" field, see below screenshot.
+
+![Codespaces settings](./docs/images/Codespaces_settings.png)
+
+When you click "Create", a new Codespace instance is created with a development environment based on the Docker container that is defined in this repo's Dockerfile. Recommended plug-ins are also automatically installed for you.
+
+1. Done! You can now start developing using Q# in your browser. Open a terminal to start running your first samples (see [here](#first-samples)).
+
+### Visual Studio Code ###
+
+If you prefer to develop code locally, we recommend to install an editor such as [Visual Studio Code](https://code.visualstudio.com/download). To get started in VS Code, simply install the editor from [here](https://code.visualstudio.com/download). Download this repository to your computer and open the folder in VS Code. The editor will automatically recognize the files in the `.vscode` folder and request you to install the recommended extension. This includes the [Microsoft Quantum Development Kit for Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=quantum.quantum-devkit-vscode) extension which is the fastest way to get started with the QDK.
+
+Open a terminal to start running your first samples (see [here](#first-samples)).
+
+
+### Running a Jupyter Notebook with Docker
```

Comment:
Should this mention that the `.devcontainer/devcontainer.json` file provided with this repo automatically integrates the Docker container with VS Code?

Reply:
I'm not sure this is within the scope of this PR. I think it would make more sense to add that to our Docs.

Reply:
Fair enough, I guess my only thought was that the discussion is specific to how to use this repo rather than the Quantum Development Kit as a whole. It may be helpful even just to link to existing docs at https://github.com/microsoft/Quantum/blob/master/.devcontainer/README.md.

Reply:
Sure, I think it would be really helpful to document it somewhere, including the links to the right docs to use. I just think that this is not in scope for this specific PR :)